zerver/actions/users.py

```diff
@@ -221,54 +140,84 @@ def do_delete_user_preserving_messages(user_profile: UserProfile) -> None:
             full_name=f"Deleted User {user_id} (temp)",
             active=False,
             is_mirror_dummy=True,
-            force_date_joined=date_joined,
+            force_date_joined=user_profile.date_joined,
             create_personal_recipient=False,
         )
-        # Uses index: zerver_message_realm_sender_recipient (prefix)
-        Message.objects.filter(realm_id=realm.id, sender=user_profile).update(
-            sender=temp_replacement_user
+
+        overwrite_data = model_to_dict(
+            temp_replacement_user,
+            exclude=[
+                "id",
+                "date_joined",
+                "realm",
+                "recipient",
+                # TODO: Do we want to keep original uuid for any reason, or overwrite?
+                "uuid",
+                # TODO: what about last_login? might be useful to preserve, but perhaps might leak identity?
+                "last_login",
```

Comment:
I think `last_login` and `date_joined` probably fall into the same category as `date_joined`; basic metadata about this account's period of activity. There's no personal data there.

Reply:
yeah i agree. removing these `TODO` notes then.